consensus/src/simplex/prover.rs

```diff
@@ -502,4 +503,45 @@ mod tests {
         );
         assert!(result.is_none());
     }
+
+    #[test]
+    fn test_serialize_deserialize_aggregation() {
+        // Create a proposal to sign
+        let (view, parent, payload) = (1, 0, test_digest(0));
+        let proposal = wire::Proposal {
+            view,
+            parent,
+            payload: payload.to_vec(),
+        };
+        let message = proposal_message(view, parent, &payload);
+
+        // Create a proof with 3 signers
+        const NAMESPACE: &[u8] = b"test";
+
+        // Reproduceable test
+        use rand::SeedableRng;
+        let mut rng = rand::rngs::StdRng::seed_from_u64(3);
```

Comment:
this fails reliably w/o fix, at least on my machine

Reply:
Would you mind importing `rand::SeedbalRng` at the top of the test block (line 443)?

Reply:
By convention, we also use seed 0 for tests (if you could change that)?

Reply:
It does not fail with 0 on my machine (w/o fix). I can still change it if you like?

Reply:
Turns out it was failing (parsing the wrong public keys). I added a few more assertions to make sure parsing passes and we get the right info out.